### ISSUE TITLE
Clarification on Auxiliary Prompt Implementation (alignment loss)

### ISSUE THREAD

**Issue by KosarKeshavarz** [chunmeifeng/SPRC#17]
I have a question about the implementation of the **auxiliary prompt** as described in the paper (Eq. 2 & 3).

#### **In the paper**:

You define the auxiliary prompt $p'_i$ as being optimized per sample using a contrastive loss (Eq. 2 & 3).
This is used in the alignment loss and it makes total sense since it has guidance capabilities and can be considered as per sample ideal prompt.

#### **In the code**:

It seems that instead of computing $p'_i$ per sample, you use a shared `prompt_tokens` vector:

```python
loss_align = F.mse_loss(fusion_output.mean(1), prompt_tokens.clone().detach().mean(1))
```

#### **My question**:

1. Is the per-sample optimization of the auxiliary prompt implemented elsewhere in the code? ( blip2_cir_align_prompt ?)
2. If not, what is the reasoning behind using the shared `prompt_tokens` vector instead? 
3. How does this simplification impact the alignment loss and theoretical justification?

Thanks!
